charts/bindplane/templates/transform-agent.yaml

```diff
@@ -0,0 +1,65 @@
+apiVersion: apps/v1
+kind: {{ include "bindplane.deployment_type" . }}
+metadata:
+  name: bindplane-transform-agent
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: bindplane-transform-agent
+    app.kubernetes.io/stack: bindplane
+    app.kubernetes.io/component: transform-agent
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  {{- if eq (include "bindplane.deployment_type" .) "StatefulSet" }}
+  replicas: 1
+  serviceName: bindplane-transform-agent
+  {{- end }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: bindplane-transform-agent
+      app.kubernetes.io/stack: bindplane
+      app.kubernetes.io/component: transform-agent
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: bindplane-transform-agent
+        app.kubernetes.io/stack: bindplane
+        app.kubernetes.io/component: transform-agent
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      serviceAccountName: default
+      securityContext:
+        fsGroup: 65534
+      containers:
+        - name: transform-agent
+          image: ghcr.io/observiq/bindplane-transform-agent:latest
```

Comment:
I think we should stick the image and tag into the values file and set defaults there. There is a pattern for this already, without defaults because the bindplane image's version is derived from the Chartfile.